action/protocol/execution/evm/contract.go

```diff
@@ -174,7 +160,7 @@ func (c *contract) Snapshot() Contract {
 }
 
 // newContract returns a Contract instance
-func newContract(addr hash.Hash160, account *state.Account, dao db.KVStore, batch db.CachedBatch, opts ...ContractOption) (Contract, error) {
+func newContract(addr hash.Hash160, account *state.Account, dao db.KVStore, opts ...ContractOption) (Contract, error) {
```

Comment:
opts ...ContractOption is added to pass in HistoryRetentionOption()

since HistoryRetentionOption() is removed, opts can be removed as well, also the definition of ContractOption